tests/unit/test_services.py

```diff
@@ -840,3 +840,44 @@ def test_fix_doesnt_fail_on_empty_file() -> None:
     result = fix_code(source)
 
     assert result == source
+
+
+def test_file_that_only_has_unused_imports() -> None:
+    """
+    Given: A file that only has unused imports.
+    When: Fix code is run.
+    Then: The output should be a single empty line.
+    """
+    source = dedent(
+        """\
+        import os
+        import sys
+        """
+    )
+
+    result = fix_code(source)
+
+    assert result == "\n"
+
+
+def test_file_that_only_has_unused_imports_with_no_qa() -> None:
+    """
+    Given: A file that has unused imports, but one marked with noqa.
+    When: Fix code is run.
+    Then: The output should be only the import marked with noqa.
+    """
+    source = dedent(
+        """\
+        import os  # noqa: F401
```

Comment:
The line should also be removed, as it's not using `# noqa: autoimport`, so you've found another bug xD

Reply:
I'd suggest to either change `F401` to `autoimport`, or even better, if you have time and want to, check why `_remove_unused_imports` is not working as expected and fix it

Reply:
Sorry for taking so long in answering, it's not being a good month for me.

`autoimport` does more things than flake8/pyflakes, for example moving imports to the top, that behaviour is not covered by any existing annotation, that's why we introduced our own. I agree with you that it should respect flake8 annotations though, so I'd say you've found even another bug xD.

If you analyze the flow of the test with `pdb` you'll see that pyflakes is showing errors on both imports, (even the one that has the `F401`). Put a trace between lines 246 and 247 of `src/autoimport/model.py`, maybe it adds a property or something where we can tell apart the errors that have been ignored with annotations

Reply:
Hey @fsouza were you able to take a look at my comment?

Reply:
Oh sorry, yeah so that noqa is read by flake8 I think, not pyflakes. I'll remove that test.

Reply:
Rebased the PR and removed that test. I think it should be good to go now.